kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -179,6 +179,23 @@ private static Optional<Predicate> constructDataSkippingFilter(
                     return e2Filter; // possibly none
                 }
 
+            // Match any file whose null count is less than the row count.
+            case "IS_NOT_NULL":
+                Expression child = getUnaryChild(dataFilters);
+                if (child instanceof Column) {
+                    Column childColumn = (Column) child;
+                    if (schemaHelper.isSkippingEligibleNullCountColumn((Column) child)) {
+                        return Optional.of(
+                            new Predicate(
+                                "<",
+                                schemaHelper.getNullCountColumn(childColumn),
+                                schemaHelper.getNumRecordsColumn()
```

Comment:
does this take into account the number of deleted rows due to DVs? Or it shouldn't change the null count remains same? What about the case where the `tightBounds` is true? Is the null count remains same?

I think it would be good to add few tests with DVs and tightBounds=true. wdyt?

Reply:
I will add a test but I looked into it and this should be fine.

When rows are deleted with DV tightBounds is set to false, but the rowCount AND nullCount remains the same (is not updated). https://github.com/delta-io/delta/blob/master/PROTOCOL.md#per-file-statistics

So when tightBounds=false, if nullCount=numRecords --> all rows must be null and we can skip since the original file had all nulls.

Reply:
Added tests